assets/src/design-system/components/radio/radio.js

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useMemo } from 'react';
+import styled, { css } from 'styled-components';
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * Internal dependencies
+ */
+import { themeHelpers, THEME_CONSTANTS } from '../../theme';
+import { FOCUS_VISIBLE_SELECTOR } from '../../theme/global';
+import { Text } from '../typography';
+
+const BORDER_WIDTH = 1;
+const TEXT_GAP_HEIGHT = 4;
+const RING_DIAMETER = 22;
+const BUTTON_CONTAINER_HEIGHT = RING_DIAMETER + BORDER_WIDTH * 2;
+const CONTAINER_HEIGHT = BUTTON_CONTAINER_HEIGHT + TEXT_GAP_HEIGHT;
+const RADIO_DIAMETER = 16;
+
+const Container = styled.div`
+  display: flex;
+  align-items: center;
+  height: ${CONTAINER_HEIGHT}px;
+`;
+
+const ButtonContainer = styled.div`
+  position: relative;
+  height: ${BUTTON_CONTAINER_HEIGHT}px;
+  width: ${BUTTON_CONTAINER_HEIGHT}px;
+`;
+
+const LabelContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  margin-left: 12px;
```

Comment:
this might need to be reworked a little for spacing on RTL. 
![Screen Shot 2021-03-10 at 9 58 32 AM](https://user-images.githubusercontent.com/10720454/110667175-400b9100-8187-11eb-83b1-49788684d810.png)